proposals/0010-resource-element-type-validation.md

```diff
@@ -0,0 +1,217 @@
+* Proposal: [0010](0010-resource-element-type-validation.md)
+* Author(s): [Joshua Batista](https://github.com/bob80905)
+* Sponsor: Joshua Batista
+* Status: **Under Consideration**
+* Impacted Project(s): (LLVM)
+* Issues: [#75676](https://github.com/llvm/llvm-project/issues/75676)
+
+## Introduction
+Resources are often used in HLSL, with various resource element types.
+
+For example:
+```
+RWBuffer<float> rwbuf: register(u0);
+```
+In this code, the element type is `float`, and the resource type is `RWBuffer`.
+`RWBuffer`, along with some other buffers and textures, fall under the "typed buffer"
+category. Other buffer resources are categorized as "raw buffers". 
+Below is a description of resources that are considered "typed buffers" vs "raw buffers".
+* typed buffers
+  * [RW|RasterizerOrdered]Buffer
+  * [Feedback]Texture*
+* raw buffers
+  * [Append|Consume|RW|RasterizerOrdered]StructuredBuffer
+  * [RW]ByteAddressBuffer
+
+There is a distinct set of rules that define valid element types for typed buffer resources
+and valid element types for raw buffer resources.
+
+Element types for typed buffer resources:
+* Are not intangible (e.g., isn't a resource type)
+* Must be vectors or scalars of arithmetic types, not bools nor enums nor arrays
+* Should be a scalar or homogenous vector of a floating-point or integer type, with a maximum of 4 components after translating 64-bit components into pairs of uint32_t components
+Element types for raw buffer resources:
+* Are not intangible (e.g., isn't a resource type)
+* All constituent types must be arithmetic types or bools or enums

```

Comment:
Other than the language nit of "Element types for raw buffer resources", is there anything that's tangible which isn't allowed?  If not I don't think we need the next point.

If we still need this point, does "arithmetic types or bools or enums" sound inclusive of aggregates of such types?  It's not clear to me that it is.

Reply:
I believe we are disallowing arrays. The second point doesn't sound inclusive of this, I'll fix this.

Reply:
I don't think there's anything that's tangible which isn't allowed. I've removed the second point.